packages/web/stores/user-settings/show-dust.tsx

```diff
@@ -0,0 +1,39 @@
+import { action, makeObservable, observable, computed } from "mobx";
+import { FunctionComponent } from "react";
+import { Switch } from "../../components/control";
+import { IUserSetting } from ".";
+
+export type ShowDustState = { showDust: boolean };
+
+export class ShowDustUserSetting implements IUserSetting<ShowDustState> {
+  readonly id = "show-dust";
+  readonly displayLabel: string;
+  readonly controlComponent: FunctionComponent<ShowDustState> = ({
+    showDust,
+  }) => (
+    <Switch
+      isOn={showDust}
+      onToggle={() => this.setState({ showDust: !showDust })}
+    />
+  );
```

Comment:
Yeah this is not common, haha. But I felt it needed to be included here to keep all code related to a user setting in one place. Only this implementation knows what control UI is appropriate for the state it is encapsulating. Perhaps as an alternative we could create a typescript interface or delegate to pass props to some component?

My goal is to isolate all code related to a user setting in one object/place, to avoid having user setting logic scattered throughout the app. The exception being checking the setting to chage behavior in the app.

Reply:
I'm just concerned that although it is not a big problem in terms of performance, components will be always in memory, and the consistency with other store logic is broken. I think `ShowDustUserSetting` just can have a string as a component type like `"switch"` and not be concerned about how the component is drawn.

Reply:
I think since the state is a generic param, I'm just going to keep it there. It also helps isolate all the details about how a user setting is represented to the user. At the end of the day, tsx compiles to React.createElement JS, so I don't think there's a performance problem.

Reply:
I understand what you mean. just feel unfamiliar because I've been thinking if it's common to separate the state and view(component). but I agree your opinion too.